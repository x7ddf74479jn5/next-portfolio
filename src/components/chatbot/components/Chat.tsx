import Avatar from "@material-ui/core/Avatar";
import ListItem from "@material-ui/core/ListItem";
import ListItemAvatar from "@material-ui/core/ListItemAvatar";
import React from "react";

import type { Chat as ChatType } from "../../../types/chatbot/index";
import { icons } from "../../../utils/images";

type Props = ChatType;

const Chat: React.VFC<Props> = (props) => {
  const isQuestion = props.type === "question";
  const classes = isQuestion ? "p-chat__row" : "p-chat__reverse";

  return (
    <ListItem className={classes}>
      <ListItemAvatar>
        {isQuestion ? (
          <Avatar alt="icon" src={icons.pandashark.path} />
        ) : (
          <Avatar alt="icon" src={icons.noProfile.path} />
        )}
      </ListItemAvatar>
      <div className="p-chat__bubble">{props.text}</div>
    </ListItem>
  );
};

export default Chat;
