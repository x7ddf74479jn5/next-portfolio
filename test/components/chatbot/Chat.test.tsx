import Chat from "src/components/chatbot/Chat";
import type { RenderResult } from "test/test-utils";
import { reTestCase } from "test/test-utils";
import { render } from "test/test-utils";

describe("Chat", () => {
  let renderResult: RenderResult;

  describe("question", () => {
    beforeEach(() => {
      renderResult = render(<Chat text="text" type="question" />);
    });

    it("matches snapshot", () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it("rendered correctly", () => {
      expect(renderResult.getByRole("listitem")).toBeInTheDocument();
      expect(renderResult.getByRole("listitem")).toHaveClass("chatRow");
      expect(renderResult.getByText("text")).toHaveClass("chatBubble");
      expect(renderResult.getByRole("img")).toHaveAttribute("src", reTestCase.anyImage);
      expect(renderResult.getByRole("img")).toHaveAttribute("alt", "icon");
    });
  });

  describe("answer", () => {
    beforeEach(() => {
      renderResult = render(<Chat text="text" type="answer" />);
    });

    it("matches snapshot", () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it("rendered correctly", () => {
      expect(renderResult.getByRole("listitem")).toBeInTheDocument();
      expect(renderResult.getByRole("listitem")).toHaveClass("chatReverse");
      expect(renderResult.getByText("text")).toHaveClass("chatBubble");
      expect(renderResult.getByRole("img")).toHaveAttribute("src", reTestCase.anyImage);
      expect(renderResult.getByRole("img")).toHaveAttribute("alt", "icon");
    });
  });
});
