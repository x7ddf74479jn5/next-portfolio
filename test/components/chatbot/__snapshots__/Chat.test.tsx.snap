// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat answer matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="MuiListItem-root chatReverse MuiListItem-gutters"
  >
    <div
      class="MuiListItemAvatar-root"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular"
      >
        <img
          alt="icon"
          class="MuiAvatar-img"
          src="/img/icons/no-profile.png"
        />
      </div>
    </div>
    <div
      class="chatBubble"
    >
      text
    </div>
  </li>
</DocumentFragment>
`;

exports[`Chat question matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="MuiListItem-root chatRow MuiListItem-gutters"
  >
    <div
      class="MuiListItemAvatar-root"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular"
      >
        <img
          alt="icon"
          class="MuiAvatar-img"
          src="/img/icons/pandashark_180-180.png"
        />
      </div>
    </div>
    <div
      class="chatBubble"
    >
      text
    </div>
  </li>
</DocumentFragment>
`;
